### ISSUE TITLE
Dynamically Pass a div id to Chartist.Pie

### ISSUE THREAD

**Issue by goldesigner** [chartist-js/chartist#441]
Hello.
Thanks for the wonderful library, very easy to use for noobs like me!
I am using Chartist in a Meteor app to create multiple pie charts on a page. I am running Chartist as a template helper function  and targeting the div id to render the chart .  When i try to pass a div id  to Chartist as a string variable, I get the error below:

```
  Uncaught TypeError: Cannot read property 'querySelectorAll' of null
```

Here's my  code. 
I have a "report" template that iterates over Chartdb database and renders a "chart" template for each object in the database. 

_model.js_

```
Chartdb = new Meteor.Collection("chartdb"); 
// contains objects in the format - { chartData : [4, 2, 4, 2] }
```

_client.js_

```
Template.report.helpers({
    charts : function() {
      return Chartdb.find().fetch();
    },
    chartDraw : function() {
        var divId = "#" + this._id;
        new Chartist.Pie(divId, {
            series: this.chartData      
            }, {
            donut: false,
            showLabel: false

        });
    }
});
```

_"report" template:_

```
<template name="report">
   <div class="container">
        {{#each charts}}
            {{> chart}}
            {{chartDraw}}
        {{/each}}

  </div>
</template>
```

_"chart" template:_

```
<template name="chart">
    <div class="ct-chart ct-square" id="{{_id}}"></div>
</template>
```

**Comment by gionkunz** [comment]
I'm not a meteor expert but the error says in a very weird way (we don't provide too many custom errors)  that the element with the selector you've specified was not found. Can you check against the DOM if the element really exists already at that stage? 

I remember there were other issues concerning Meteor and probably they were also related to rendering and timing. Maybe you can search for them?

I'm sorry that I couldn't be that of a help here. Please let me know if you need more information from the Chartist.js side.

Cheers
Gion

**Comment by goldesigner** [comment]
Thanks for looking into this. I had to switch to another library since I couldn't fix it in time. 

I did have the issue with rendering and timing as you mention. it is mainly because Meteor documentation for the `Template.templatename.onRendered()` method is not clear. The function has to be passed as an argument to OnRendered. This pattern is different from `Template.templatename.helpers` and `Template.templatename.events`, where a function is passed as a javascript object.
